frontend/apps/pwa/src/features/recorder/FieldEditor.tsx

```diff
@@ -1,26 +1,57 @@
-import { FieldDefinition } from 'core-api-client';
-import React, { FC, Fragment } from 'react';
+import { FieldDefinition, SelectOption } from 'core-api-client';
+import React, { FC } from 'react';
 
 import { RecordPickerContainer } from '../../components/RecordPicker';
 
 import { FormValue } from './recorder.slice';
 
 export type FieldEditorProps = {
   field: FieldDefinition;
-  value: any;
-  setValue: (value: any) => void;
+  value: string | string[] | null;
+  setValue: (value: string | string[] | null) => void;
   addSubRecord: () => void;
   selectSubRecord: (subRecordId: string) => void;
   subRecords: FormValue[] | undefined;
 };
 
+export const mapFieldDescription = (fd: FieldDefinition) => {
+  if (fd.description) {
+    return <small className="text-muted">{fd.description}</small>;
+  }
+  return <></>;
+};
+export const mapFieldLabel = (fd: FieldDefinition) => {
+  return (
+    <label className="form-label opacity-75" htmlFor={fd.id}>
+      {fd.name}
+    </label>
+  );
+};
+
+export const mapSelectOptions = (required: boolean, key: boolean, options?: SelectOption[]) => {
+  if (!options) {
+    return <></>;
+  }
+  return (
+    <>
+      <option aria-label="no value" disabled={required || key} value="" />
+      {options.map((o) => (
```

Comment:
```suggestion
      {options?.map((o) => (
```

Reply:
This will not happen as we have a check above for falsy options